src/render/program/raster_program.js

```diff
@@ -64,7 +74,10 @@ const rasterUniformValues = (
     'u_saturation_factor': saturationFactor(layer.paint.get('raster-saturation')),
     'u_contrast_factor': contrastFactor(layer.paint.get('raster-contrast')),
     'u_spin_weights': spinWeights(layer.paint.get('raster-hue-rotate')),
-    'u_perspective_transform': perspectiveTransform
+    'u_perspective_transform': perspectiveTransform,
+    'u_colorization_mix': colorMapInputMix,
+    'u_colorization_scale': colorScaleFactors(colorRange),
```

Comment:
Nit: it would be nice to avoid setting these two to NaNs if there's no colorization, just to make sure there's zero overhead for usual raster layers.

Reply:
Good catch. I think it's the case that uniforms will not be set if they are not specified in the shader. See:

https://github.com/mapbox/mapbox-gl-js/blob/81981ae18d5ea8015203415143083fafd49bb8ee/src/render/uniform_binding.js#L53

To that end, this commit moved the `uniform` definitions _inside_ `#ifdef RASTER_COLOR`, which I think will be sufficient to prevent this uniform from being set when raster color is not active. 
 c4fbb785973a9646221c0f0bd0107ee904efe61f